runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -136,15 +130,14 @@ public CheckpointWriter(CorfuRuntime rt, UUID streamId, String author, T map) {
      * @return Token at which the snapshot for this checkpoint was taken.
      */
     public Token appendCheckpoint() {
-        // Queries the sequencer for the stream tail and global log tail.
-        // The global log tail will provide the snapshot for this checkpoint,
-        // while the stream tail will provide the last observed update to this stream at
-        // the given snapshot.
-        TokenResponse response = rt.getSequencerView().query(new UUID[]{streamId});
-        // The checkpoint snapshot is given by the Global Tail
-        Token snapshot = response.getToken();
+        // We enforce a NO_OP entry for every checkpoint, i.e., a hole with backpointer map info,
```

Comment:
Can we use LogData.EMPTY instead of a hole because I think a hole by definition should not have back pointers.

Reply:
That might fail the replication layer code, because it checks for empty and tries to fill those slots.